### PR TITLE
Fs - the new file system code is in

### PR DESCRIPTION
**Pull request by pkivolowitz**
Uses new database too.

### CODE REVIEW

File: server/db_component.cpp [hunk 1]
```diff
@@ -228,9 +230,10 @@ int DB::IntegerQuery(string & sql)
 	return rv;
 }
 
-int DB::GetTrackCount()
+int DB::GetTrackCount(string nspace)
```
Comment (LOZORD) [inline server/db_component.cpp:233]: Maybe leave a comment saying that `nspace` has already been properly escaped/cleaned?

File: server/db_component.cpp [hunk 2]
```diff
@@ -322,9 +325,9 @@ bool DB::IsAColumn(std::string c)
 	return rv;
 }
 
-int DB::GetArtistCount()
+int DB::GetArtistCount(string nspace)
```
Comment (LOZORD) [inline server/db_component.cpp:328]: Same as above.

File: server/db_component.cpp [hunk 3]
```diff
@@ -333,36 +336,70 @@ bool DB::Initialized()
 	return connection != nullptr;
 }
 
-string DB::PathFromID(unsigned int id, string * title, string * artist)
+string DB::PathFromID(unsigned int id, string * title, string * artist, string nspace)
```
Comment (LOZORD) [inline server/db_component.cpp:339]: Shouldn't `nspace` be before the pointers according to the style guide?

File: server/db_component.cpp [hunk 4]
```diff
@@ -333,36 +336,70 @@ bool DB::Initialized()
...
 			if (artist != nullptr)
 				*artist = res->getString("artist");
+			rv = res->getString("fname");
 		}
+		cerr << *title << endl;
+		cerr << *artist << endl;
 	}
 	catch (LoggedException s)
 	{
+		throw s;
```
Comment (LOZORD) [inline server/db_component.cpp:370]: If the `stmt == nullptr` is the only call that will cause a `LoggedException` you can remove that check from the try-catch.

**Review by LOZORD — CHANGES_REQUESTED** [review]

File: server/db_component.cpp (hunk 1 above)
Comment (pkivolowitz) [reply server/db_component.cpp:233]: Time for a pass through the code to comment.

Comment (pkivolowitz) [reply server/db_component.cpp:233]: Apparently I have more changes to me.

File: server/db_component.cpp (hunk 3 above)
Comment (pkivolowitz) [reply server/db_component.cpp:339]: nspace has a default value - therefore it must be at the end. In c/c++ once you go default, you can't go back.

Comment (LOZORD) [reply server/db_component.cpp:339]: Good point; fine with me in that case.

**Comment by pkivolowitz** [comment]
I am cancelling this because it is incomplete. I neglected to update the database accesses for some of the other commands such as select. Be back some day soon.

**Pull request closed without merging**